### PR TITLE
Add products grid scaling setting

### PR DESCRIPTION
**Pull request by Mbucari**
This was a bitch and a half to get right, but I'm really happy with the results.

Things inside:

### Add WinForms DPI migration to remove stored form sizes
[This display bug](https://github.com/rmcrackan/Libation/issues/666#issuecomment-1629372938) was caused by old column width settings. I added a migration to High DPI that removes all stored sizes from the settings.

### Add Grid Scaling Settings

Two new settings: `GridScaleFactor` and `GridFontScaleFactor`.

`GridScaleFactor` changes row height and scales images.
`GridFontScaleFactor` just changes the grid text font.
Both range from 0.5 to 2

### Add textbox clear button

For @CharlieRussel because I like him and it wasn't too hard.

### CODE REVIEW

File: Source/LibationWinForms/ClearableTextBox.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Drawing;
+using System.Windows.Forms;
+
+namespace LibationWinForms
+{
+	public partial class ClearableTextBox : UserControl
```
Comment (rmcrackan) [inline Source/LibationWinForms/ClearableTextBox.cs:7]: Interesting. I assumed you would have inherited from TextBox

**Comment by rmcrackan** [comment]
Wow, this is a friggin huge one! Thank you so much. This must have taken forever.

**Comment by rmcrackan** [comment]
Ready for full release?

**Comment by Mbucari** [comment]
> Wow, this is a friggin huge one! Thank you so much. This must have taken forever.

If I had domesticated I'd say it took about 16 hours. But it was Damned frustrating. Compounding that frustration is coming up with two entirely different solutions for scaling to completely different frameworks. This is dual UI is annoying.

Ready for full release?

Yessir!

**Comment by rmcrackan** [comment]
For the sake of release notes, what are the github issue/ticket numbers?

**Comment by rmcrackan** [comment]
I've got these:

* settings for scaling to accommodate high DPI displays
* Download high quality cover art
* Add full context menu to call columns
* Add 'Re-download' context menu item
* improve startup time

**Comment by Mbucari** [comment]
Including the failed 10.5.2

#669
#666
#661
#663
#662 (forced re-download and context menus on all cells)

**Comment by rmcrackan** [comment]
Is this close enough?

* #666 : settings for scaling to accommodate high DPI displays
* #669 : button to clear filter
* #661 : Download high quality cover art
* #662 : Add full context menu to call columns
* #662 : Add 'Re-download' context menu item
* #663 : fix titles
* improve startup time

**Comment by Mbucari** [comment]
Replace "Download high quality cover art" with "embed high quality cover art in audio files"

But everything else is good

**Pull request merged**